pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -218,6 +218,27 @@ runCmd() {
     "$@"
 }
 
+getNixpkgsPath() {
+    nixpkgsPath=
+
+    # Get NixOS configuration
+    NIXOS_CONFIG=${NIXOS_CONFIG:-$(runCmd nix-instantiate --find-file nixos-config)}
```

Comment:
nixos-rebuild fails here if `NIXOS_CONFIG` environment variable is not set and `nixos-config` nix path is not set. (This is a case for my system)

Reply:
I thought `nixos-config=/etc/nixos/configuration.nix` was in the default NIX_PATH. Does `nixos-rebuild edit` not work for you?

Reply:
It didn't make sense to use this environment variable or nix path when I introduced buidling system by attrset in pr 320462, when I was inspired by nix-build arguments.

> Does `nixos-rebuild edit` not work for you?

yes, both `nixos-rebuild edit` and `nixos-rebuild edit --file /path/to/nixos/config/entry-point.nix` does not work.

Reply:
So, should I add a default /etc/nixos/configuration.nix? Or what?

Reply:
if getting $NIXOS_CONFIG is empty and nix-config from nix-path fails, don't execute code where $NIXOS_CONFIG is used.

Reply:
Done.

Reply:
not resolved, execution in line 225 can still fail
```suggestion
    NIXOS_CONFIG=${NIXOS_CONFIG:-$(runCmd nix-instantiate --find-file nixos-config 2> /dev/null || true)}
```